phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java

```diff
@@ -621,55 +623,96 @@ public void testMultiTenantImmutableTableMetadata() throws Exception {
-            connection.createStatement()
-                    .execute("create table " + tableName
-                            + " (k VARCHAR PRIMARY KEY, v1 VARCHAR, v2 VARCHAR) "
-                            + "UPDATE_CACHE_FREQUENCY=" + tableSpecificUpdateCacheFrequency);
-            rs = connection.createStatement().executeQuery(readSysCatQuery);
-            Assert.assertTrue(rs.next());
-            Assert.assertEquals(tableSpecificUpdateCacheFrequency, rs.getLong(2));
-        } finally {
-            if (connection != null) {
-                connection.createStatement().execute("drop table if exists " + tableName);
-                connection.close();
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        String createTableString = "CREATE TABLE " + tableName + " (k VARCHAR PRIMARY KEY,"
+                + "v1 VARCHAR, v2 VARCHAR)";
+        verifyUCFValueInSysCat(tableName, createTableString, props, 0L);
+    }
+
+    @Test
+    public void testCreateTableWithTableLevelUpdateCacheFreq() throws Exception {
+        String tableName = generateUniqueName();
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+
+        HashMap<String, Long> expectedUCF = new HashMap<>();
+        expectedUCF.put("10", new Long(10L));
+        expectedUCF.put("0", new Long(0L));
+        expectedUCF.put("10000", new Long(10000L));
+        expectedUCF.put("ALWAYS", new Long(0L));
+        expectedUCF.put("NEVER", new Long(Long.MAX_VALUE));
+
+        for (HashMap.Entry<String, Long> entry : expectedUCF.entrySet()) {
+            String tableLevelUCF = entry.getKey();
+            long expectedUCFInSysCat = entry.getValue();
+
+            String createTableString = "CREATE TABLE " + tableName + " (k VARCHAR PRIMARY KEY,"
+                    + "v1 VARCHAR, v2 VARCHAR) UPDATE_CACHE_FREQUENCY = " + tableLevelUCF;
+            verifyUCFValueInSysCat(tableName, createTableString, props, expectedUCFInSysCat);
+        }
+    }
+
+    @Test
+    public void testCreateTableWithInvalidTableUpdateCacheFreqShouldThrow() throws Exception {
+        String tableName = generateUniqueName();
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+
+        ArrayList<String> invalidUCF = new ArrayList<>();
+        invalidUCF.add("GIBBERISH");
+        invalidUCF.add("10000.6");
+
+        for (String tableLevelUCF : invalidUCF) {
+            String createTableString = "CREATE TABLE " + tableName + " (k VARCHAR PRIMARY KEY,"
+                    + "v1 VARCHAR, v2 VARCHAR) UPDATE_CACHE_FREQUENCY = " + tableLevelUCF;
+            try {
+                verifyUCFValueInSysCat(tableName, createTableString, props, -1L);
+                fail();
+            } catch (IllegalArgumentException e) {
+                // expected
+                assertEquals("Table's UPDATE_CACHE_FREQUENCY can only be set to 'ALWAYS', 'NEVER' or a millisecond numeric value.",
```

Comment:
I dislike comparing actual string messages in the exception, since a subtle change can cause test failures. Is there any other way you can confirm this via a sqlCode comparison for example?

Reply:
The message comparison was probably an overhead in the first place, I only need to verify that an IllegalArgumentException was thrown. Will remove the message comparison.